frontend/src/models/devices.ts

```diff
@@ -1,213 +1,122 @@
-import fuzzy from 'fuzzy'
-import Device from '../services/Device'
-import { IDevice, IService } from 'remote.it'
+import { graphQLFetch, graphQLAdaptor } from '../services/graphQL'
 import { createModel } from '@rematch/core'
-import { renameServices } from '../helpers/nameHelper'
-import { updateConnections } from '../helpers/connectionHelper'
-import { IRawDevice } from 'remote.it'
 import { r3 } from '../services/remote.it'
 
-// Slightly below the API limit for search of 300 services.
-const SEARCH_ONLY_SERVICE_LIMIT = 300
+type DeviceParams = { [key: string]: any }
 
-const SORT_SETTING_KEY = 'sort'
-const SEARCH_ONLY_SETTING_KEY = 'search-only'
-
-interface DeviceState {
+type IDeviceState = DeviceParams & {
   all: IDevice[]
-  searchPerformed: boolean
-  fetched: boolean
+  total: number
+  results: number
+  searched: boolean
   fetching: boolean
-  searchOnly: boolean
-  searching: boolean
   destroying: boolean
   query: string
-  sort: SortType
-}
-
-async function fetchAll(cache: boolean = true): Promise<IDevice[]> {
-  const [allDevices, metadata] = await Promise.all([
-    r3.get(`/device/list/all?cache=${cache.toString()}`).then(({ devices }: { devices: IRawDevice[] }) => devices),
-    r3.devices.metadata(),
-  ])
-  return r3.devices.group(allDevices, metadata)
+  append: boolean
+  filter: 'all' | 'active' | 'inactive'
+  size: number
+  from: number
 }
 
-const state: DeviceState = {
+const state: IDeviceState = {
   all: [],
-  searchPerformed: false,
-  fetched: false,
+  total: 0,
+  results: 0,
+  searched: false,
   fetching: true,
-  searching: false,
-  searchOnly: false,
   destroying: false,
   query: '',
-  sort: (window.localStorage.getItem(SORT_SETTING_KEY) || 'state') as SortType,
+  append: false,
+  filter: 'all',
+  size: 50,
+  from: 0,
 }
 
 export default createModel({
   state,
   effects: (dispatch: any) => ({
-    /**
-     * Decide if the user should only search for devices veruses
-     * us fetching all of their devices at the getgo.
-     */
-    async shouldSearchDevices() {
-      // First see if they have already decided on their preference
-      const pref = window.localStorage.getItem(SEARCH_ONLY_SETTING_KEY)
-      let searchOnly = false
-
-      // Handle unset state - localStorage turns bool into string
-      if (typeof pref === 'string') {
-        searchOnly = pref === 'true'
+    async fetch(_, globalState: any) {
+      const { set, graphQLProcessor } = dispatch.devices
+      const { query, filter, size, from, append } = globalState.devices
+      const options: gqlOptions = {
+        size,
+        from,
```

Comment:
Maybe use the infinite pagination with last/after

{
  login {
    devices(size: 5, after: "80:00:00:00:01:04:8B:59") {
      hasMore
      last
      items {
        name
      }
    }
  }
}